### PR TITLE
feat(files): blur nsfw image thumbnails based on settings

### PR DESCRIPTION
**Pull request by josephmcg**
**What this PR does** 📖
- allow nsfw uploads in filesystem with nsfw boolean on each file
- if file is nsfw and nsfw filter is on (settings), apply blur effect to image

**Which issue(s) this PR fixes** 🔨
AP-1431

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Comment by phillsatellite** [comment]
When tested I had toggle off and NSFW image would still have blur over it. I created new accounts and tried again with toggle on and then again with toggle off and each time blur would still be on. Verified on Chrome and Safari

https://user-images.githubusercontent.com/93608357/164759910-faf5c79e-af0a-4487-ad60-2ab5be1007fc.mov

**Comment by josephmcg** [comment]
This is primarily concerned with file system uploads, chat should be handled separately

**Review by phillsatellite — APPROVED** [review]

File: components/views/files/file/File.vue [hunk 1]
```diff
@@ -47,7 +47,7 @@ export default Vue.extend({
     }
   },
   computed: {
-    ...mapState(['ui']),
+    ...mapState(['ui', 'settings']),
```
Comment (molimauro) [inline components/views/files/file/File.vue:50]: Can you extract only the state needed? :)

File: components/views/files/view/View.vue [hunk 2]
```diff
@@ -27,7 +27,7 @@ export default Vue.extend({
     }
   },
   computed: {
-    ...mapState(['ui']),
+    ...mapState(['ui', 'settings']),
```
Comment (molimauro) [inline components/views/files/view/View.vue:30]: Same

**Review by molimauro — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Pull request merged**